packages/core/src/system.ts

```diff
@@ -0,0 +1,34 @@
+import { ActorSystem, ActorSystemInfo, AnyActorRef } from './types.js';
+
+export function createSystem<T extends ActorSystemInfo>(): ActorSystem<T> {
+  let sessionIdIndex = 0;
+  const children = new Map<string, AnyActorRef>();
+  const keyedActors = new Map<keyof T['actors'], AnyActorRef | undefined>();
+  const reverseKeyedActors = new WeakMap<AnyActorRef, keyof T['actors']>();
+
+  const system: ActorSystem<T> = {
+    _register: (actorRef) => {
+      const id = `x:${sessionIdIndex++}`;
+      children.set(id, actorRef);
+      return id;
+    },
+    _unregister: (actorRef) => {
+      children.delete(actorRef.id);
+      const key = reverseKeyedActors.get(actorRef);
+
+      if (key !== undefined) {
+        keyedActors.delete(key);
+        reverseKeyedActors.delete(actorRef);
```

Comment:
this one could be called unconditionally, otoh - it kinda doesn't matter cause you are using `WeakMap`, in a way... there is no point in this `.delete` call at all, the engine itself should just clean this up - that's the whole point of the Weak part in the WeakMap